tests/test_cmd2.py

```diff
@@ -1278,6 +1283,18 @@ def help_app():
     return HelpApp()
 
 
+def test_help_headers(capsys) -> None:
+    help_app = HelpApp()
+    help_app.onecmd_plus_hooks('help')
+    out, err = capsys.readouterr()
+
+    assert help_app.doc_leader in out
```

Comment:
For these asserts to really be testing something useful, I think you need to substantially update the `HelpApp` class to actually set those headers.  Currently it does not.

Reply:
cmd2 was already setting these members, but I went ahead and explicitly set them in HelpApp just to be extra careful.